### PR TITLE
Error parser tests

### PR DESCRIPTION
**Pull request by sbayer55**
Signed-off-by: Steven Bayer <smbayer@amazon.com>

### Description
Added simple parser tests
 
### Issues Resolved
Missing parser tests
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParserTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    @Test
+    void testInvalidNumberOfOperands() {
+        assertThatHasParseError("5==");
+        assertThatHasParseError("==");
+        assertThatHasParseError("not");
+    }
+
+    @Test
+    void testValidOperatorsRequireSpace() {
```
Comment (dlvenable) [inline data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParserTest.java:100]: These tests could be made into `@ParameterizedTest`s using `@ValueSource`. But, they should still be fine this way.

```
@ParameterizedTest
@ValueSource(strings = {
  "/status in {200}",
  "/a==(/b==200)",
  ...
})
void testValidOperatorsRequireSpace(String expression) {
  assertThatIsValid(expression);
}
```

File: data-prepper-expression/src/main/antlr/DataPrepperExpression.g4 [hunk 2]
```diff
@@ -225,27 +225,91 @@ EscapeSequence
...
-OR : 'or';
-NOT : 'not';
-LPAREN : '(';
-RPAREN : ')';
-LBRACE : '{';
-RBRACE : '}';
-FORWARDSLASH : '/';
-DOUBLEQUOTE : '"';
-SET_SEPARATOR : ',';
+SET_DELIMITER
```
Comment (dlvenable) [inline data-prepper-expression/src/main/antlr/DataPrepperExpression.g4:228]: Is there a particular reason to use the extra spacing here?

**Review by dlvenable — DISMISSED** [review]

File: data-prepper-expression/src/main/antlr/DataPrepperExpression.g4 (hunk 2 above)
Comment (sbayer55) [reply data-prepper-expression/src/main/antlr/DataPrepperExpression.g4]: I was expecting to make the lexer rules more complex when adding support for required space operators. I have reverted the extra spacing.

File: data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParserTest.java (hunk 1 above)
Comment (sbayer55) [reply data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParserTest.java]: Great idea on the `@ParameterizedTest`. It gives the benefits of `assertAll(...)` with more descriptive error messages!

**Review by dlvenable — DISMISSED** [review]
Nice! Thanks!

**Review by asifsmohammed — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Pull request merged**